metaflow/plugins/kfp/kfp_step_function.py

```diff
@@ -28,11 +53,15 @@ def kfp_step_function(
         shell=True,
         universal_newlines=True,
         executable="/bin/bash",
-        env=dict(
-            os.environ,
-            METAFLOW_USER="kfp-user",  # TODO: what should this be for a non-scheduled run?
-            METAFLOW_SERVICE_URL=metaflow_service_url,
-        ),
+        env={
+            "METAFLOW_DATASTORE_SYSROOT_S3": datastore_root,
+            "METAFLOW_SERVICE_URL": metaflow_service_url,
+            "PRECEDING_COMPONENT_INPUTS": json.dumps(preceding_component_inputs),
+            "PRECEDING_COMPONENT_OUTPUTS": json.dumps(preceding_component_outputs),
+            "METAFLOW_USER": "kfp-user",  # TODO: what should this be for a non-scheduled run?
+            **preceding_component_outputs_env,
+            **os.environ,
```

Comment:
Just to confirm here you want to swap the ordering so `os.environ` takes precedence over `preceding_component_outputs_env` and the hard coded keys? Makes sense to me but I think that was different to the implementation in the prior commit and the version above :)

Reply:
Thank you for the catch!